src/Uno.UI/Microsoft/UI/Xaml/Controls/TreeView/TreeViewViewModel.cs

```diff
@@ -0,0 +1,856 @@
+		//	var tvnNewNode = get_self<TreeViewNode>(newNode);
+		//	m_collectionChangedEventTokenVector.insert(m_collectionChangedEventTokenVector.begin() + index, tvnNewNode.ChildrenChanged({ this, &ViewModel.TreeViewNodeVectorChanged }));
+		//	m_IsExpandedChangedEventTokenVector.insert(m_IsExpandedChangedEventTokenVector.begin() + index, tvnNewNode.AddExpandedChanged({ this, &ViewModel.TreeViewNodePropertyChanged }));
+		//}
+
+		//void ViewModel.RemoveAt(uint32_t index)
+		//{
+		//	var inner = GetVectorInnerImpl();
+		//	var current = inner.GetAt(index).as< TreeViewNode > ();
+		//	inner.RemoveAt(index);
+
+		//	// Unhook event handlers
+		//	var tvnCurrent = get_self<TreeViewNode>(current);
+		//	tvnCurrent.ChildrenChanged(m_collectionChangedEventTokenVector[index]);
+		//	tvnCurrent.RemoveExpandedChanged(m_IsExpandedChangedEventTokenVector[index]);
+
+		//	// Remove tokens from vectors
+		//	m_collectionChangedEventTokenVector.erase(m_collectionChangedEventTokenVector.begin() + index);
+		//	m_IsExpandedChangedEventTokenVector.erase(m_IsExpandedChangedEventTokenVector.begin() + index);
+		//}
+
+		private void Append(object value)
+		{
+			GetVectorInnerImpl().Add(value);
+			TreeViewNode newNode = (TreeViewNode)value;
+
+			// Hook up events and save tokens
+			var tvnNewNode = newNode;
+			//TODO
+			//m_collectionChangedEventTokenVector.push_back(tvnNewNode.ChildrenChanged({ this, &ViewModel.TreeViewNodeVectorChanged }));
+			//m_IsExpandedChangedEventTokenVector.push_back(tvnNewNode.AddExpandedChanged({ this, &ViewModel.TreeViewNodePropertyChanged }));
+		}
+
+		private void RemoveAtEnd()
+		{
+			var inner = GetVectorInnerImpl();
+			var current = (TreeViewNode)inner[Count - 1];
+			inner.RemoveAt(inner.Count - 1);
+
+			// unhook events
+			var tvnCurrent = current;
+			//TODO
+			//tvnCurrent.ChildrenChanged(m_collectionChangedEventTokenVector.back());
+			//tvnCurrent.RemoveExpandedChanged(m_IsExpandedChangedEventTokenVector.back());
+
+			// remove tokens
+			//TODO
+			//m_collectionChangedEventTokenVector.pop_back();
+			//m_IsExpandedChangedEventTokenVector.pop_back();
+		}
+
+		private new void Clear()
+		{
+			// Don't call GetVectorInnerImpl().Clear() directly because we need to remove hooked events
+			int count = Size;
+			while (count != 0)
+			{
+				RemoveAtEnd();
+				count--;
+			}
+		}
+
+		private void ReplaceAll(object[] items)
+		{
+			var inner = GetVectorInnerImpl();
+			inner.Clear();
+			inner.AddRange(items);
+		}
+
+		//// Helper function
+		internal void PrepareView(TreeViewNode originNode)
+		{
+			// Remove any existing RootNode events/children
+			var existingOriginNode = m_originNode;
+			if (existingOriginNode != null)
+			{
+				for (int i = (existingOriginNode.Children.Count - 1); i >= 0; i--)
+				{
+					var removeNode = existingOriginNode.Children[i];
+					RemoveNodeAndDescendantsFromView(removeNode);
+				}
+
+				//TODO
+				//if (m_rootNodeChildrenChangedEventToken.value != 0)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)